docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,300 @@
+and creates a new wallet with the selected members,
+instructing them to begin key generation.
+The selected members may contain duplicate operators.
+The new wallet is added to the pending wallets.
+
+The submitter of this sortition transaction is rewarded.
+
+==== Sortition transaction reimbursement
+
+When TBTC is collected in deposit fees,
+a certain fraction (e.g. 10%) of the fee is placed in the sortition fee pool.
+
+The transaction to perform a new sortition can be submitted by anyone.
+The submitter of the transaction is reimbursed from the sortition fee pool.
+If at least 24h has elapsed since the most recent sortition,
+the full contents of the sortition fee pool are released to the submitter.
+If less than 24h has elapsed,
+the amount released to the submitter is reduced,
+and the remainder is released as general rewards
+to other participants in the TBTC system.
+
+This reduction can be linear
+(the fraction released equals the fraction of 24h that has passed),
+quadratic/cubic
+(the fraction released equals the square/cube of the time passed)
+or exponential
+(for every hour before 24h has passed, the released fraction is halved).
+
+This lets the market decide when enough reward has accumulated
+to cover the gas fees of a new sortition transaction
+without the need for methods such as price oracles.
+It also balances the creation rate of new wallets with the deposit rate;
+when large amounts of BTC are deposited,
+new wallets will be created more frequently.
+
+By limiting the reward before 24h has elapsed,
+excessive creation of new wallets is disincentivised.
+This aims to give each wallet a reasonable opportunity to collect deposits
+before it is replaced by a new one,
+and limit the number of wallets active at the same time.
+
+=== Key generation
+
+Because every participant needs to be present and active
+for the full duration of key generation,
+failures are likely to happen.
+To avoid expensive new sortition,
+a two-phase retry protocol is specified to deal with failures.
+
+In this RFC, the word "majority" means at least _T_ out of _N_ operators,
+if _T_ is greater than _N/2+1_.
+
+==== Two-phase retry protocol
+
+The full key generation protocol consists of individual _attempts_.
+In each attempt,
+the specified participants try to perform DKG
+with the protocol from `tss-lib`.
+If the attempt fails due to one or more participants being inactive,
+the inactive participants are removed from the DKG participants
+and another attempt is made.
+
+If the number of DKG participants falls below the critical size _C_,
+no further attempt is made by the remaining participants.
+Instead, the group will clear the list of inactive participants
+and wait 12h before making another attempt with the entire group.
+This repeats until 7 days have elapsed
+since the timestamp of the sortition transaction.
+If this ultimate timeout is reached without succesful key generation,
+the group dissolves and makes no further attempts.
```

Comment:
Yes, that's the idea and that's why we can afford the 7 day ultimate timeout. It creates some of the cross-cutting concerns earlier mentioned wrt the ordering of wallets and dealing with multiple key generations, but it isn't particularly difficult to address especially if we add some sanity checks (e.g. a maximum number of concurrently active DKGs so that someone can't spend lots of gas money to grief the system by creating too many pending wallets)